widgets/common/src/ManageLabelsModal.jsx

```diff
@@ -38,13 +40,17 @@ function ManageLabelsModal({ deploymentId, open, onHide, toolbox }) {
         clearErrors();
         setLoading();
 
-        actions.doSetLabels(deploymentId, labels).then(onHide).catch(setMessageAsError).finally(unsetLoading);
+        actions
+            .doSetLabels(deploymentId, [...existingLabels, ...labels])
+            .then(onHide)
+            .catch(setMessageAsError)
+            .finally(unsetLoading);
     }
 
     return (
         <Modal open={open} onClose={onHide}>
             <Modal.Header>
-                <Icon name="tags" /> {i18n.t('widgets.common.labels.modalHeader', { deploymentId })}
+                <Icon name="tags" /> {header || i18n.t('widgets.common.labels.modalHeader', { deploymentId })}
```

Comment:
1. Maybe we can have two headers in `en.json` - e.g. one starting from 'Manage', the second one starting from 'Add' and use only `existingLabels` prop to determine which one to choose?
2. Maybe it would be also good to change the button name for add-mode to be 'Add' instead of 'Apply'?

Reply:
Could we do

```suggestion
                <Icon name="tags" /> {header ?? i18n.t('widgets.common.labels.modalHeader', { deploymentId })}
```

to allow setting `header` to an empty string?

Reply:
> Could we do
> ...
> to allow setting `header` to an empty string?

I don't think an empty header would be a valid case.

Reply:
> 1. Maybe we can have two headers in `en.json` - e.g. one starting from 'Manage', the second one starting from 'Add' and use only `existingLabels` prop to determine which one to choose?

We could, but the effect of setting `existingLabels` would be a little bit surprising then.
Regarding strings organization, I would leave it as it is for now and refine it in the future - If I understand correctly this modal is going to be used in at least one more place. Once it is used everywhere where it's intended to it will be clear which strings are indeed common, and which are specific to particular widgets. It will be easier to accurately organize the strings then.

> 2. Maybe it would be also good to change the button name for add-mode to be 'Add' instead of 'Apply'?

Done.